packages/tooltip/src/vaadin-tooltip-overlay.js

```diff
@@ -27,6 +27,58 @@ class TooltipOverlay extends PositionMixin(OverlayElement) {
 
     return memoizedTemplate;
   }
+
+  static get properties() {
+    return {
+      position: {
+        type: String,
+        reflectToAttribute: true,
+      },
+    };
+  }
+
+  /**
+   * @protected
+   * @override
+   */
+  _updatePosition() {
+    super._updatePosition();
+
+    if (!this.positionTarget) {
+      return;
+    }
+
+    // Center the tooltip overlay horizontally
+    if (this.position === 'bottom' || this.position === 'top') {
+      const targetRect = this.positionTarget.getBoundingClientRect();
+      const overlayRect = this.$.overlay.getBoundingClientRect();
+
+      const offset = targetRect.width / 2 - overlayRect.width / 2;
+
+      if (this.style.left) {
+        const left = parseFloat(this.style.left) + offset;
```

Comment:
nit: could probably use `overlayRect.left` here to avoid parsing

Reply:
Thanks, done.

Reply:
The `parseFloat(this.style.top)` below could also be changed but looks like `parseFloat(this.style.right)` is still needed tho?

Reply:
Replaced `parseFloat(this.style.top)` with `overlayRect.top`. As for `this.style.right`, yes we need to parse it, otherwise we would have to essentially calculate it once again, which I would like to avoid.